homeassistant/components/sensor/dyson.py

```diff
@@ -183,3 +204,101 @@ def state(self):
         if self._device.environmental_state:
             return self._device.environmental_state.volatil_organic_compounds
         return None
+
+
+class DysonParticulateMatter25Sensor(DysonSensor):
+    """Representation of Dyson pm25 sensor."""
+
+    def __init__(self, device):
+        """Create a new Dyson pm25 sensor."""
+        super().__init__(device, 'particulate_matter')
+        self._name = "{} Particulate matter 2.5 μg/m3"\
+            .format(self._device.name)
+
+    @property
+    def state(self):
+        """Return pm25 level."""
+        if self._device.state:
+            return int(self._device.environmental_state.particulate_matter_25)
+        return None
+
+
+class DysonParticulateMatter10Sensor(DysonSensor):
+    """Representation of Dyson pm10 sensor."""
+
+    def __init__(self, device):
+        """Create a new Dyson pm10 sensor."""
+        super().__init__(device, 'particulate_matter')
+        self._name = "{} Particulate matter 10 μg/m3".format(self._device.name)
+
+    @property
+    def state(self):
+        """Return pm10 level."""
+        if self._device.state:
+            return int(self._device.environmental_state.particulate_matter_10)
+        return None
+
+
+class DysonNitrogenDioxideSensor(DysonSensor):
+    """Representation of Dyson no2 sensor."""
+
+    def __init__(self, device):
+        """Create a new Dyson no2 sensor."""
+        super().__init__(device, 'nitrogen_dioxide')
+        self._name = "{} Nitrogen dioxide".format(self._device.name)
+
+    @property
+    def state(self):
+        """Return no2 level."""
+        if self._device.state:
+            return int(self._device.environmental_state.nitrogen_dioxide)
+        return None
+
+
+class DysonVolatileOrganicCompoundsSensor(DysonSensor):
```

Comment:
What kind of measurement is this? Does it also fit as air quality?

Reply:
It is indeed an air quality measurement, same as the Nitrogen Dioxide and Particulate Matter sensors.
Unfortunately for both the Volatile Compounds and Nitrogen Dioxide values, it's hard to say what the units should be - in my experience of using this platform over the last week, I haven't seen values for either over 100 so have assumed it's intended as a %.

Reply:
Dyson seems to either measure VOC and NO2 in µg/m³ or convert to that unit because their Air Quality value is a max(PM2.5, PM10, VOC, NO2), so all four scales have to be normalized. The scale that you'll find in the Dyson Link app (0-3, 4-6, 7-8, 9+) looks like 1/10 of the api values.